stringfuzz/transformers/translate.py

```diff
@@ -0,0 +1,36 @@
+'''
+Permuting the alphabet in every string literal.
+'''
+
+import random
+import string
+
+from stringfuzz.ast import StringLitNode
+from stringfuzz.ast_walker import ASTWalker
+from stringfuzz.parser import parse
+
+__all__ = [
+    'translate',
+]
+
+class TranslateTransformer(ASTWalker):
+    def __init__(self, ast):
+        super(TranslateTransformer, self).__init__(ast)
+        shuffled = list(string.ascii_letters)
```

Comment:
The class `string.ascii_letters` doesn't cover a lot of characters in our string literals. The character list that the scanner is using is `[^"]` though, which is pretty huge. We should at least include numbers, since those appear pretty often in literals, but I'm not sure about the entirety of "not a double quote".

What do you think? Should we do what the scanner does, or pick some in-between?

Reply:
Ifaz mentions that 

> The permuation must fix the digit characters pointwise. You can instead disallow the integer-to-string and string-to-integer operators from Z3str programs.

I went with string.ascii_letters to avoid numbers, and because string literals appear in StrToReNodes so other characters can have special meanings that will also mess with the sat guarantees.

Reply:
I'd say this is an ideal case for adding parameters (or even a preprocessor!) to the transformer. In the mode with IntToStr and StrToInt we move digits, and in the other we do not. Also, what do you think of special characters, like dashes and question marks?